gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2324,15 +2514,31 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
         if( poFieldDefn->GetWidth() == 1 )
         {
             nType = NC_CHAR;
-            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1,
+
+            if(!m_bLegacyCreateMode)
+            {
+                nVarID = m_poDS->vcdf.nc_def_vvar(pszVarName, nType, 1, &nMainDimId);
+            }
+
+            else
+            {
+                status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1,
                                 &nMainDimId, &nVarID);
+            }
         }
 #ifdef NETCDF_HAS_NC4
         else if( m_poDS->eFormat == NCDF_FORMAT_NC4 && m_bUseStringInNC4 )
         {
             nType = NC_STRING;
-            status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1,
-                                &nMainDimId, &nVarID);
+            if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation